lib/ftp.js

```diff
@@ -125,15 +132,18 @@ class Client {
      */
     login(user = "anonymous", password = "guest") {
         this.ftp.log(`Login security: ${describeTLS(this.ftp.socket)}`);
-        return this.ftp.handle("USER " + user, (res, task) => {
-            if (positiveCompletion(res.code)) { // User logged in proceed OR Command superfluous
+        return this.ftp.handle("USER " + user, (err, res, task) => {
+            if (err) {
+                task.reject(err);
+            }
+            else if (positiveCompletion(res.code)) { // User logged in proceed OR Command superfluous
                 task.resolve(res);
             }
             else if (res.code === 331) { // User name okay, need password
                 this.ftp.send("PASS " + password);
             }
             else { // Also report error on 332 (Need account)
-                task.reject(res);
+                task.reject(new Error(`Unexpected response from FTP server: ${res.message}`));
```

Comment:
Shouldn't this be an `FTPError`,?  Per the new README: "Any error reported by the FTP server will be thrown as `FTPError`"
( I would expect that as long as there's a `code` of any sort, that would manifest as an FTPError)